node/src/reactor/main_reactor/upgrade_shutdown.rs

```diff
@@ -114,17 +118,21 @@ impl MainReactor {
             .signature_gossip_tracker
             .finished_gossiping_enough(validator_weights);
         if finished_gossiping_enough {
-            // Allow a delay to acquire more finality signatures
-            let effects = effect_builder
-                .set_timeout(DELAY_BEFORE_SHUTDOWN)
-                .event(|_| MainEvent::ControlAnnouncement(ControlAnnouncement::ShutdownForUpgrade));
-            // should not need to crank the control logic again as the reactor will shutdown
-            UpgradeShutdownInstruction::Do(DELAY_BEFORE_SHUTDOWN, effects)
+            self.shutdown(effect_builder)
         } else {
             UpgradeShutdownInstruction::CheckLater(
                 "waiting for completion of gossiping signatures".to_string(),
                 DELAY_BEFORE_SHUTDOWN,
             )
         }
     }
+
+    fn shutdown(&self, effect_builder: EffectBuilder<MainEvent>) -> UpgradeShutdownInstruction {
```

Comment:
nit: rename to `schedule_shutdown_for_upgrade` or something similar to reflect the fact that this is upgrade specific logic.